notebooks/setfit-absa-fiqa.ipynb

```diff
@@ -0,0 +1,7780 @@
+{
```

Comment:
<p>[SetFitABSA](<a href="https://huggingface.co/blog/setfit-absa)" target="_blank">https://huggingface.co/blog/setfit-absa)</a></p><p>is a framework for <strong>**few-shot**</strong> training of domain-specific ABSA models. It provides an efficient and accurate technique to detect sentiment towards specific aspects within the text. The following graph shows evaluation results of a SetFitABSA model that was trained with just few k finance related sentences and tested on the same 235 FiQA_SA test sentences that were used used to evaluate BloombergGPT. The size of the model is based on its underlying sentence transformer and is usually considerably smaller compared to SOTA LLMs.</p><p>Specifically, in this work, we used the <strong>**paraphrase-mpnet-base-v2**</strong> sentence transformer which contains only <strong>**110M**</strong> params compared to <strong>**50B**</strong> of BloombergGPT.&lt;br&gt;</p><p> </p><p>We separated from each prompt in [FiQA_SA](<a href="https://huggingface.co/datasets/AdaptLLM/finance-tasks/viewer/FiQA_SA)" target="_blank">https://huggingface.co/datasets/AdaptLLM/finance-tasks/viewer/FiQA_SA)</a> the tagged and non-tagged sentences, to create [train and test sets](<a href="https://huggingface.co/datasets/ronenlap/SetFitAbsa_FiQA)" target="_blank">https://huggingface.co/datasets/ronenlap/SetFitAbsa_FiQA)</a> correspondingly. The train set contains 646 unique sentences overall.&lt;br&gt;This graph summarizes the evaluation results. It presents the weighted F1-score achieved by SetFitABSA as a function of k, the number of sampled training sentences. The result for each k is averaged over 5 seeds. We can see that with only 50 tranining sentences SetFitABSA exceeds the BloombergGPT score. BTW, if we use the full training set for training the SetFitABSA model we get a weighted F1-score of over 86.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/setfit/pull/471/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>